front-dispatcher/src/features/DashboardPage/components/FilterOnSmallDevices/FilterOnSmallDevices.tsx

```diff
@@ -1,40 +1,33 @@
 import React, { useRef, useState } from "react";
 
 import { apiStrings, searchBarStrings } from "../../../../strings/strings";
-import { FilterBtnDiv, FilterDiv, SortByDiv } from "./style";
+import { SortByDiv } from "./style";
 import { ReactComponent as DropIcon } from "../../../../assets/dropdown.svg";
 import { ReactComponent as FilterIcon } from "../../../../assets/filter.svg";
 import { Backdrop } from "@mui/material";
-import PrimaryButton from "../../../../components/PrimaryButton/style";
+import FilterContent from "./components/FilterContent";
 const FilterOnSmallDevices = () => {
   const [clicked, setClicked] = useState<boolean>();
-  const searchBtn = useRef<HTMLButtonElement | null>(null);
+  const backDropElem = useRef(null);
   return (
     <>
       <SortByDiv onClick={() => setClicked(!clicked)}>
         <div>
-          {apiStrings.EverythingOptions[0]}
+          {apiStrings.Everything[0]}
           <DropIcon style={{ marginLeft: "1em" }} />
         </div>
         <FilterIcon />
       </SortByDiv>
       {clicked && (
         <Backdrop
+          ref={backDropElem}
           sx={{ color: "#fff", zIndex: "5" }}
           open={clicked}
           onClick={(e) => {
-            if (searchBtn && e.target == searchBtn.current) return;
+            if (backDropElem && e.target !== backDropElem.current) return;
```

Comment:
why the return is empty?

Reply:
on click function only happens if the element clicked is the backdrop element